### DISCUSSION TITLE
What is the best way to copy/duplicate an Item Stack?

### DISCUSSION THREAD

**Discussion by skies-unbound** [Minestom/Minestom#1395] [Help] [answered]
Currently I am working on the crafting stations feature such as workbench (not vanilla). I want to create an copy of origin item stack and save it into a task instance. For instance, a recipe consumes 24 iron ingots, and I submit a stack of 64. What is the best way to split them into 24 and 40 in code?
I am using
`ItemStack.fromNBT(
     ingredient.material(),
     ingredient.toItemNBT(),
     recipe.amount()
)`
now.
Is there any other solution?

**Comment by ZakShearman** [comment, marked as answer]
You can simply use ItemStack#withAmount. ItemStacks are immutable so performing any action like withAmount will create a copy with the new amount. There shouldn't be a situation in which you need to make an identical copy.

Might be just me but I don't get exactly how ur using it, but here's just an example i guess.

```java
ItemStack initialItem = ItemStack.builder(Material.BEDROCK).amount(64).build();

int usedAmount = 20;

ItemStack reducedItem = initialItem.withAmount(initialItem.amount() - reducedAmount);
```

**Reply by skies-unbound** [reply]
I get it, but does it mean that I cannot set the properties of an existing item stack? ItemStack#withAmount would not modify current stack but returns a new stack with modified amount, right? If there is an item stack in the inventory, I have to set the returned value from ItemStack#withAmount to the slot, is that true?

**Reply by ZakShearman** [reply]
Yes that is correct.
